processfamily/test_processfamily.py

```diff
@@ -24,86 +27,249 @@
 from processfamily import _traceback_str
 import signal
 import threading
+import pytest
+from pytest_lazyfixture import lazy_fixture
 
 from processfamily.futurecompat import get_env_dict, list_to_native_str
 
 if sys.platform.startswith('win'):
     from processfamily._winprocess_ctypes import CAN_USE_EXTENDED_STARTUPINFO, CREATE_BREAKAWAY_FROM_JOB
+    import win32service
+    import win32serviceutil
+    from processfamily.test.ExeBuilder import build_service_exe
+    from processfamily.processes import USE_PROCESS_QUERY_LIMITED_INFORMATION
 
-class _BaseProcessFamilyFunkyWebServerTestSuite(unittest.TestCase):
+pid_dir = os.path.join(os.path.dirname(__file__), 'test', 'tmp', 'pid')
+path_to_ParentProcessPy = os.path.join(os.path.dirname(__file__), 'test', 'ParentProcess.py')
 
-    skip_crash_test = None
 
-    def setUp(self):
-        self.pid_dir = os.path.join(os.path.dirname(__file__), 'test', 'tmp', 'pid')
-        if not os.path.exists(self.pid_dir):
-            os.makedirs(self.pid_dir)
-        for pid_file in self.get_pid_files():
-            with open(pid_file, "r") as f:
-                pid = f.read().strip()
-            if pid and self.process_exists_or_access_denied(int(pid)):
-                logging.warning(
-                    ("Process with pid %s is stilling running. This could be a problem " + \
-                    "(but it might be a new process with a recycled pid so I'm not killing it).") % pid )
-            else:
-                os.remove(pid_file)
-        self.check_server_ports_unbound()
+def process_exists_or_access_denied(pid):
+    try:
+        return process_exists(pid)
+    except AccessDeniedError as e:
+        # It is most likely that this process does exist!
+        return True
+
+
+def kill_process_ignore_access_denied(pid):
+    try:
+        return kill_process(pid)
+    except AccessDeniedError as e:
+        # Can't do anything about this
+        pass
 
-    def process_exists_or_access_denied(self, pid):
+
+def assert_middle_child_port_unbound():
+    port = Config.get_starting_port_nr() + 2
+    logging.info("Checking for ability to bind to port %d", port)
+    serversocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    try:
+        #if not sys.platform.startswith('win'):
+            # On linux I need this setting cos we are starting and stopping things
+            # so frequently that they are still in a STOP_WAIT state when I get here
```

Comment:
Can you delete the part of the comment that reference windows and linux. I think that just makes it confusing.